src/rpc.rs

```diff
@@ -333,30 +312,30 @@ pub fn generate_app_specific_cells(
 	index: u32,
 	max_col: u16,
 	num: u64,
-	block: Block,
+	header: Header,
 	id: u32,
 ) -> Vec<Cell> {
 	let mut buf = Vec::new();
-	let hash_ind = get_id_specific_size(block);
+	let hash_ind = get_id_specific_size(header);
 	let endsize = hash_ind.get(&id).unwrap();
 
-	for index in index..=*endsize {
-		let rows = (index - 1) as u16 / max_col;
-		let cols = (index - 1) as u16 % max_col;
+	(index..*endsize).for_each(|index| {
+		let rows = (index) as u16 / max_col;
+		let cols = (index) as u16 % max_col;
 
 		buf.push(Cell {
 			block: num,
 			row: rows as u16,
 			col: cols as u16,
 			..Default::default()
 		});
-	}
+	});
 	buf
 }
 
-pub fn get_id_specific_size(num: Block) -> HashMap<u32, u32> {
```

Comment:
Maybe to rename num to header or something like that. I think that `num` has confused me at some point somewhere in a codebase :D

Reply:
yeh me too, just went with the flow. Will change